File: backend/src/main/java/com/gymapp/modules/training/dto/WorkoutPlanResponse.java
package com.gymapp.modules.training.dto;

import com.gymapp.modules.training.enums.WpType;
import lombok.Builder;
import lombok.Data;
import java.util.List;
import java.util.UUID;

@Data
@Builder
public class WorkoutPlanResponse {
    private UUID id;
    private String name;
    private String description;
    private WpType planType;
    private String assignedToName;
    private String createdByName;
    private List<PlanExerciseResponse> exercises;
}

File: android/app/src/main/java/com/gymapp/android/data/remote/response/WorkoutPlanResponse.kt
package com.gymapp.android.data.remote.response

import com.gymapp.android.domain.model.workout.WpType

data class WorkoutPlanResponse(
    val id: String,
    val name: String,
    val description: String?,
    val planType: String?,
    val targetLevel: String? = null,
    val assignedToName: String? = null,
    val createdByName: String? = null,
    val scheduledDate: String? = null,
    val exercises: List<PlanExerciseResponse> = emptyList()
)